src/main/groovy/com/google/protobuf/gradle/GenerateProtoTask.groovy

```diff
@@ -637,13 +667,50 @@ public abstract class GenerateProtoTask extends DefaultTask {
 
   protected String computeExecutablePath(ExecutableLocator locator) {
     if (locator.path != null) {
-      return locator.path
+      return locator.path.endsWith(JAR_SUFFIX) ? createJarTrampolineScript(locator.path) : locator.path
     }
     File file = locatorToAlternativePathsMapping.getting(locator.name).get().singleFile
+    if (file.name.endsWith(JAR_SUFFIX)) {
+      return createJarTrampolineScript(file.getAbsolutePath())
+    }
+
     if (!file.canExecute() && !file.setExecutable(true)) {
       throw new GradleException("Cannot set ${file} as executable")
     }
     logger.info("Resolved artifact: ${file}")
     return file.path
   }
+
+  /**
+   * protoc only supports plugins that are a single self contained executable file. For .jar files create a trampoline
+   * script to execute the jar file. Assume the jar is a "fat jar" or "uber jar" and don't attempt any artifact
+   * resolution.
+   * @param jarAbsolutePath Absolute path to the .jar file.
+   * @return The absolute path to the trampoline executable script.
+   */
+  private String createJarTrampolineScript(String jarAbsolutePath) {
+    assert jarAbsolutePath.endsWith(JAR_SUFFIX)
+    boolean isWindows = isWindows()
+    File scriptExecutableFile = new File("${project.buildDir}/scripts/" +
+            jarAbsolutePath[0..(jarAbsolutePath.length() - JAR_SUFFIX.length())] + "-trampoline." +
+            (isWindows ? "bat" : "sh"))
+    try {
+      if (createNewScriptFile(scriptExecutableFile)) {
+        String javaExe = computeJavaExePath(isWindows)
+        // Rewrite the trampoline file unconditionally (even if it already exists) in case the dependency or versioning
+        // changes we don't need to detect the delta (and the file content is cheap to re-generate).
+        new FileOutputStream(scriptExecutableFile).withCloseable { execOutputStream ->
+          String trampoline = isWindows ?
+                  "@ECHO OFF\r\n${javaExe} -jar ${jarAbsolutePath} %*\r\n" :
```

Comment:
All of these variables need to be quoted. On Windows especially it will be normal to have spaces in the directory name. Was Java not installed into "Program Files" when testing?

For Windows we may just quote the string and hope for the best. Since the string does not end in a backslash, that will mostly work. For Unix we can use `'` and do a string replacement beforehand to `s/'/'\''/` (or `s/'/'"'"'/` if you want to have more fun!).

Reply:
good catch. I did some brief research on windows path escaping and found `%` needs to be escaped, so I covered that.

Reply:
validated changes with java located in the "Program Files" dir: `C:\Program Files\AdoptOpenJDK\jdk-8.0.242.08-hotspot\jre\bin\java.exe`